packages/apollo-server-core/src/caching.ts

```diff
@@ -0,0 +1,12 @@
+export interface KeyValueCache {
+  set(
+    key: string,
+    data: string,
+    options?: {
+      ttl?: number;
+      tags?: string[];
+    },
+  ): Promise<void>;
+  get(key: string): Promise<string>;
+  invalidate(tags: string[]): Promise<void>;
```

Comment:
Should we call this `invalidateTags` to avoid confusion with key-based invalidation?

Reply:
I like it! Let's do that